plugins/channel-safely/channel-safely-plugin.cpp

```diff
@@ -0,0 +1,211 @@
+/* Prevent channeling down into known open space.
+Author:  Josh Cooper
+Created: Aug. 4 2020
+Updated: Jun. 29 2021
+*/
+
+#include <PluginManager.h>
+#include <modules/EventManager.h>
+#include "channel-safely.h"
+
+using namespace DFHack;
+
+DFHACK_PLUGIN("channel-safely");
+DFHACK_PLUGIN_IS_ENABLED(enabled);
+REQUIRE_GLOBAL(world);
+//#define CS_DEBUG 1
```

Comment:
instead of compile-time constants and undocumented "debug" commands, consider using DFHack's logging framework in `Debug.h`

Reply:
I'll try to take a look at it sometime soon.